yjit/src/core.rs

```diff
@@ -411,43 +411,101 @@ impl From<Opnd> for YARVOpnd {
     }
 }
 
+/// Number of registers that can be used for stack temps
+pub const MAX_TEMP_REGS: usize = 5;
+
 /// Maximum index of stack temps that could be in a register
 pub const MAX_REG_TEMPS: u8 = 8;
 
-/// Bitmap of which stack temps are in a register
-#[derive(Copy, Clone, Default, Eq, Hash, PartialEq, Debug)]
-pub struct RegTemps(u8);
+/// A stack slot or a local variable. u8 represents the index of it (<= 8).
+#[derive(Copy, Clone, Eq, Hash, PartialEq, Debug)]
+pub enum RegTemp {
+    Stack(u8),
+    Local(u8),
+}
+
+/// RegTemps manages a set of registers used for temporary values on the stack.
+/// Each element of the array represents each of the registers.
+/// If an element is Some, the temporary value uses a register.
+#[derive(Copy, Clone, Default, Eq, Hash, PartialEq)]
+pub struct RegTemps([Option<RegTemp>; MAX_TEMP_REGS]);
 
 impl RegTemps {
-    pub fn get(&self, index: u8) -> bool {
-        assert!(index < MAX_REG_TEMPS);
-        (self.0 >> index) & 1 == 1
+    /// Return the index of the register for a given stack value if allocated.
```

Comment:
The comment says stack value, but RegTemp suggest it could also be a local?

Reply:
My intention was that both temps and locals live on the stack so both of them are a "stack value". But I'll think of something else for clarity.

Reply:
Now that we renamed `RegTemp` to `RegOpnd`, I just called it "operand" https://github.com/ruby/ruby/pull/11157/commits/e4fe66dc1f9df5b79636edb4400be56c05b7954e